File: Unique_Number_of_Occurrences.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var uniqueOccurrences = function(arr) {
    const counts = new Map();

    for (const n of arr) {
        counts.set(n, 1 + (counts.get(n) || 0));
    }

    return counts.size === new Set(counts.values()).size;    
};

File: Unique_Number_of_Occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
       Map<Integer, Integer> counts = new HashMap<>();

        for (int n : arr) {
            counts.put(n, 1 + counts.getOrDefault(n, 0));
        }

        return counts.size() == new HashSet<>(counts.values()).size();        
    }
}
